### PR TITLE
[macOS, iOS] Sandbox: Adopt new MobileAsset endpoints

### PR DESCRIPTION
**Pull request by brentfulgham**
#### d259da0f78aebd1cd62401c82c57544c0579cfe3
<pre>
[macOS, iOS] Sandbox: Adopt new MobileAsset endpoints
<a href="https://bugs.webkit.org/show_bug.cgi?id=249352">https://bugs.webkit.org/show_bug.cgi?id=249352</a>
&lt;rdar://101780002&gt;

Reviewed by Per Arne Vollan.

Update the sandbox and entitlements for the Network Process to allow use of updates
MobileAsset endpoints when present. No change in behavior, just support for another
team refactoring code.

* Source/WebKit/NetworkProcess/mac/com.apple.WebKit.NetworkProcess.sb.in:
* Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.Networking.sb.in:
* Source/WebKit/Scripts/process-entitlements.sh:

Canonical link: <a href="https://commits.webkit.org/257959@main">https://commits.webkit.org/257959@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/5fe9c277f4320dc119d766a19c2d6c7d3f2c9ad2

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/100442 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/9602 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/33504 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/109779 "Built successfully") | 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/10514 "Built successfully") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/85/builds/215 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/92865 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/107631 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/106221 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/7960 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/91225 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/34602 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/5/builds/103957 "Passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/89876 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/22614 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/77562 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/3338 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/24133 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/3330 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/85/builds/215 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/9451 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/43623 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/5452 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/5145 "Built successfully") | | | 
| | | | |

### CODE REVIEW

File: Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.Networking.sb.in [hunk 1]
```diff
@@ -585,7 +585,9 @@
 
  ;; <rdar://problem/47598758>
 (allow mach-lookup 
-    (global-name "com.apple.nesessionmanager.content-filter"))
+    (global-name
+        "com.apple.mobileasset.autoasset"
```
Comment (pvollan) [inline Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.Networking.sb.in]: Perhaps create a separate rule for this, since the change is unrelated to <rdar://problem/47598758>?

**Review by pvollan — COMMENTED** [review]
R=me. It would be great if we could create a sandbox extension for this Mach lookup, since the UI process knows when content filtering is on. Should we file a follow-up bug on this?

File: Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.Networking.sb.in (hunk 1 above)
Comment (brentfulgham) [reply Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.Networking.sb.in]: Good point -- I will fix that before landing.

**Comment by brentfulgham** [comment]
Good idea regarding dynamic extension. I filed https://bugs.webkit.org/show_bug.cgi?id=249402 for that improvement.

**Review by pvollan — APPROVED** [review]
R=me.
